x-pack/platform/plugins/private/observability_ai_assistant_management/public/routes/components/settings_tab/change_kb_model.test.tsx

```diff
@@ -0,0 +1,235 @@
+      expect(button).toBeEnabled();
+    });
+  });
+
+  it('disables the button when knowledge base is installing', () => {
+    const mockKb = createMockKnowledgeBase({ isInstalling: true });
+    renderComponent(mockKb);
+    const button = screen.getByTestId('observabilityAiAssistantKnowledgeBaseUpdateModelButton');
+    expect(button).toBeDisabled();
+  });
+
+  // Legacy inferenceId tests cover the component behavior when currentInferenceId is LEGACY_CUSTOM_INFERENCE_ID
+  // TODO: Remove these tests after https://github.com/elastic/kibana/issues/227103 is implemented
+  const legacyInferenceTests = (inferenceId: string, label: string) => {
+    describe(`when ${inferenceId} is available`, () => {
+      beforeEach(() => {
+        setupMockGetModelOptions([
+          {
+            key: inferenceId,
+            label,
+            description:
+              'Focus on query meaning, not just keyword matching, using learned associations between terms. It delivers more relevant, context-aware results and works out of the box with no need for deep machine learning expertise.',
+          },
+          {
+            key: '.multilingual-e5-small-elasticsearch',
+            label: 'E5-small (multilingual)',
+            description:
+              'E5 is an NLP model by Elastic designed to enhance multilingual semantic search by focusing on query context rather than keywords. E5-small is a cross-platform version compatible with different hardware configurations.',
+          },
+        ]);
+      });
+
+      it('remaps legacy currentInferenceId to ELSER ID in dropdown', () => {
+        const mockKb = createMockKnowledgeBase({
+          status: createMockStatus({
+            currentInferenceId: LEGACY_CUSTOM_INFERENCE_ID,
+            endpoint: {
+              inference_id: LEGACY_CUSTOM_INFERENCE_ID,
+              task_type: 'text_embedding',
+              service: 'my-service',
+              service_settings: {},
+            },
+          }),
+        });
+        renderComponent(mockKb);
+
+        const dropdown = screen.getByTestId('observabilityAiAssistantKnowledgeBaseModelDropdown');
+        expect(dropdown).toHaveTextContent(label);
+      });
+
+      it('disables the button initially when legacy inference is active', () => {
+        const mockKb = createMockKnowledgeBase({
+          status: createMockStatus({
+            currentInferenceId: LEGACY_CUSTOM_INFERENCE_ID,
+            endpoint: {
+              inference_id: LEGACY_CUSTOM_INFERENCE_ID,
+              task_type: 'text_embedding',
+              service: 'my-service',
+              service_settings: {},
+            },
+          }),
+        });
+        renderComponent(mockKb);
+
+        const button = screen.getByTestId('observabilityAiAssistantKnowledgeBaseUpdateModelButton');
+        expect(button).toBeDisabled();
+      });
+
+      it('enables the button when user selects E5 model while legacy inference is active', async () => {
```

Comment:
```suggestion
      it('enables the button when user selects E5 model while legacy inference is active', async () => {
```

Same here. Can we make it clearer?